securesystemslib/storage.py

```diff
@@ -256,7 +256,8 @@ def getsize(self, filepath):
   def create_folder(self, filepath):
     # If called with an empty string, return immediately
     if not filepath:
-      return
+        raise securesystemslib.exceptions.StorageError(
+            "Can't create a folder with an empty filepath!")
```

Comment:
I'd rather we deleted this conditional entirely and instead improve the exception handler in `try`/`except` block around the `os.makedirs()` call below.

On Python2 `os.makedirs()` with an empty string throws an [`OSError`](https://docs.python.org/3/library/exceptions.html?#OSError), whereas on Python3 we'll get a [`FileNotFoundError`](https://docs.python.org/3/library/exceptions.html?#FileNotFoundError). In either case the exception's errno field will equal [`errno.ENOENT`](https://docs.python.org/3/library/errno.html#errno.ENOENT).

We should be able to adapt our existing exception handling to check for errno.ENOENT and pass a different message when raising `securesystemslib.exceptions.StorageError` in that case.